### PR TITLE
SONiC Yang model support for IPv6 link local

### PR DESCRIPTION
**Pull request by AkhileshSamineni**
SONiC Yang model support for IPv6 link local

What I did
Created SONiC Yang model for IPv6 link local

How I did it
Defined Yang models for IPv6 link local based on https://github.com/sonic-net/SONiC/blob/master/doc/ipv6/ipv6_link_local.md

How to verify it
Added enable test case.

Signed-off-by: Akhilesh Samineni <akhilesh.samineni@broadcom.com>

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-vlan.yang [hunk 1]
```diff
@@ -102,6 +102,16 @@ module sonic-vlan {
 					}
 				}
 
+				leaf ipv6_use_link_local_only {
+					description "Enable/Disable IPv6 link local address on vlan interface";
+					type enumeration {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-vlan.yang:107]: How about this type?
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L178

File: src/sonic-yang-models/yang-models/sonic-interface.yang [hunk 2]
```diff
@@ -75,6 +75,15 @@ module sonic-interface {
 					}
 				}
 
+				leaf ipv6_use_link_local_only {
+					description "Enable/Disable IPv6 link local address on interface";
+					type enumeration {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-interface.yang:80]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L178

File: src/sonic-yang-models/yang-models/sonic-portchannel.yang [hunk 3]
```diff
@@ -181,6 +181,16 @@ module sonic-portchannel {
 				enum disable;
 			}
 		}
+
+		leaf ipv6_use_link_local_only {
+			description "Enable/Disable IPv6 link local address on portchannel interface";
+			type enumeration {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-portchannel.yang:187]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L178

**Review by dgsudharsan — CHANGES_REQUESTED** [review]
Please update sample_config_db.json as well as Configuration guide.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/interface.json [hunk 4]
```diff
@@ -209,5 +209,33 @@
                 ]
             }
         }
+    },
```
Comment (dgsudharsan) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/interface.json:212]: Please try to cover one negative scenario (e.g.  "ipv6_use_link_local_only": "true") where yang validation should fail

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/interface.json (hunk 4 above)
Comment (AkhileshSamineni) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/interface.json:212]: Done.

File: src/sonic-yang-models/yang-models/sonic-interface.yang (hunk 2 above)
Comment (AkhileshSamineni) [reply src/sonic-yang-models/yang-models/sonic-interface.yang:80]: Done.

File: src/sonic-yang-models/yang-models/sonic-portchannel.yang (hunk 3 above)
Comment (AkhileshSamineni) [reply src/sonic-yang-models/yang-models/sonic-portchannel.yang:187]: Done.

File: src/sonic-yang-models/yang-models/sonic-vlan.yang (hunk 1 above)
Comment (AkhileshSamineni) [reply src/sonic-yang-models/yang-models/sonic-vlan.yang:107]: Done.

**Review by ganglyu — DISMISSED** [review]

**Comment by dgsudharsan** [comment]
@AkhileshSamineni  Can you please update [sample_config_db.json](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/tests/files/sample_config_db.json)  and [Configuration schema](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md)?

**Comment by AkhileshSamineni** [comment]
> @AkhileshSamineni Can you please update [sample_config_db.json](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/tests/files/sample_config_db.json) and [Configuration schema](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md)?

Done.

**Review by dgsudharsan — APPROVED** [review]

**Comment by AkhileshSamineni** [comment]
@qiluo-msft Test failure is not related to code changes. Trying the re-build again.

**Review by qiluo-msft — APPROVED** [review]

**Comment by moshemos** [comment]
@AkhileshSamineni can you fix the conflict both for 202205 as well as 202211? please priorities

**Comment by dgsudharsan** [comment]
@AkhileshSamineni Can you please create cherry-pick PR for 202205 as well?

**Comment by keboliu** [comment]
update the label since it has been included in 202211 through PR https://github.com/sonic-net/sonic-buildimage/pull/15021

**Comment by wen587** [comment]
Hi @AkhileshSamineni , could you check if it is miss that not consider sub interface case?
https://github.com/sonic-net/sonic-buildimage/issues/21631

**Pull request merged**